File: doc/src/codes/Users_Manual/stm/gain_0.cs
using AUTD3Sharp;
using AUTD3Sharp.Utils;
using AUTD3Sharp.STM;
using AUTD3Sharp.Gain;

var center = autd.Geometry.Center + new Vector3d(0, 0, 150);
const int pointNum = 200;
const double radius = 30.0;
var stm = new GainSTM(1.0).AddGainsFromIter(Enumerable.Range(0, pointNum).Select(i =>
{
    var theta = 2.0 * Math.PI * i / pointNum;
    return new Focus(center + radius * new Vector3d(Math.Cos(theta), Math.Sin(theta), 0));
}));

File: doc/src/codes/Users_Manual/stm/gain_0.py
import numpy as np
from pyautd3.gain import Focus
from pyautd3.stm import GainSTM

center = autd.geometry.center + np.array([0.0, 0.0, 150.0])
point_num = 200
radius = 30.0
stm = GainSTM(1.0).add_gains_from_iter(
    map(
        lambda theta: Focus(
            center + radius * np.array([np.cos(theta), np.sin(theta), 0])
        ),
        map(lambda i: 2.0 * np.pi * i / point_num, range(point_num)),
    )
)
